services/blockchain-service/implementations/node-blockchain-service.js

```diff
@@ -29,19 +29,47 @@ class NodeBlockchainService extends BlockchainServiceBase {
         } else {
             this[blockchainName].web3 = new Web3(blockchainRpc);
         }
+        
+        if(blockchainOptions.transactionPollingTimeout) {
+            this[blockchainName].web3.eth.transactionPollingTimeout = blockchainOptions.transactionPollingTimeout;
+        }
     }
 
     async executeContractFunction(contractName, functionName, args, blockchain) {
         const web3Instance = await this.getWeb3Instance(blockchain);
+        let result;
+        let gasPrice;
+        let transactionRetried = false;
 
-        const contractInstance = await this.getContractInstance(contractName, blockchain);
-        const tx = await this.prepareTransaction(contractInstance, functionName, args, blockchain);
-        const createdTransaction = await web3Instance.eth.accounts.signTransaction(
-            tx,
-            blockchain.privateKey,
-        );
+        while(result === undefined) {
+            try {
+                // eslint-disable-next-line no-await-in-loop
+                const contractInstance = await this.getContractInstance(contractName, blockchain);
+                // eslint-disable-next-line no-await-in-loop
+                const tx = await this.prepareTransaction(contractInstance, functionName, args, blockchain);
+                gasPrice = tx.gasPrice;
+                // eslint-disable-next-line no-await-in-loop
+                const createdTransaction = await web3Instance.eth.accounts.signTransaction(
+                    tx,
+                    blockchain.privateKey,
+                );
 
-        return web3Instance.eth.sendSignedTransaction(createdTransaction.rawTransaction);
+                // eslint-disable-next-line no-await-in-loop
+                result = await web3Instance.eth.sendSignedTransaction(createdTransaction.rawTransaction);
+            } catch(e) {
+                if (!transactionRetried && blockchain.handleNotMinedError && e.message.includes('Transaction was not mined')) {
```

Comment:
is e.message.includes('Transaction was not mined') the only way we can do this? Text of this exact message change recently if I'm not wrong so it could happen again?

Reply:
Not this message exactly, but message which appears as consequence. This is how we handle it on ot-node, but generally we can try to think of some other approach, at the moment I am not aware of better way.

Reply:
if there's no message code or something like that then I can't think of one either, it's fine like this for now, I agree